vllm/v1/worker/gpu_model_runner.py

```diff
@@ -417,6 +419,32 @@ def _prepare_inputs(self, scheduler_output: "SchedulerOutput"):
         # where M is the max_model_len.
         token_indices = (positions_np +
                          req_indices * self.input_batch.token_ids_cpu.shape[1])
+
+        # Add spec decode tokens to input_batch.token_ids_cpu.
+        # Get spec decode logits indices.
+        spec_query_end_loc = 0
+        spec_decode_logits_indices: List[int] = []
+        for i, req_id in enumerate(self.input_batch.req_ids):
+            if i == num_reqs:
+                break
+            assert req_id is not None
+            req_num_scheduled_tokens = scheduler_output.num_scheduled_tokens[
+                req_id]
+            num_compute_tokens = self.input_batch.num_computed_tokens_cpu[i]
+            spec_query_end_loc += req_num_scheduled_tokens
+            spec_token_ids: ConstantList = scheduler_output.\
+                scheduled_spec_decode_tokens.get(req_id, ConstantList([]))
```

Comment:
Can avoid creating object:
```suggestion
            spec_token_ids: ConstantList = scheduler_output.\
                scheduled_spec_decode_tokens.get(req_id) or ()
```